AmahiAnywhere/AmahiAnywhere/SettingTableViewController.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  SettingTableViewController.swift
+//  AmahiAnywhere
+//
+//  Created by Syed on 21/02/18.
+//  Copyright © 2018 Amahi. All rights reserved.
+//
+
+import UIKit
+import MessageUI
+
+class SettingTableViewController: UITableViewController ,MFMailComposeViewControllerDelegate {
+    
+    
+    //MARK:- Mail Configuration
+    
+    
+    func configureMailComposeViewController() ->MFMailComposeViewController {
+        
+        let mail = MFMailComposeViewController()
+        mail.mailComposeDelegate = self
+        mail.setToRecipients([StringLiterals.EMAIL_ADDRESS])
+        mail.setSubject(StringLiterals.SUBJECT)
+        mail.setMessageBody(StringLiterals.MESSAGE_BODY, isHTML: true)
+        
+        return mail
+    }
+    func showSendMailErrorAlert()
+    {
+        let sendMailErrorAlert = UIAlertView(title: StringLiterals.TITLE, message: StringLiterals.MESSAGE, delegate: self, cancelButtonTitle: StringLiterals.CANCLE_BUTTON_TITLE)
+        sendMailErrorAlert.show()
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+       
+        
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        
+    }
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let section = indexPath.section
+        let row = indexPath.row
+        
+        if section == 0 && row == 0{
+            
+                gotoConnections()
+             }
+        else if section == 0 && row == 1 {
+            
+               goToVersion()
+             }
+        else if section == 1 && row == 1 {
+                rateApp()
+        }
+        else if section == 1 && row == 0
+        {
+            let a = configureMailComposeViewController()
+                if MFMailComposeViewController.canSendMail()
+                {
+                    self.present(a, animated: true, completion: nil)
+
+                } else {
+                    self.showSendMailErrorAlert()
+
+                }
+            
+        }
+        else if section == 2 && row == 0 {
+            
+                logout()
+    
+        }
+        
+            print(indexPath.row)
+            tableView.deselectRow(at: indexPath, animated: true)
+        
+    }
+    
+    // MARK:- go to Connections
+    
+    func gotoConnections()
+    {
+        performSegue(withIdentifier: SegueIdentfiers.GO_TO_CONNECTION, sender: self)
+    }
+    
+    //Mark:- go to Version
+    
+    func goToVersion()
+    {
+        performSegue(withIdentifier: SegueIdentfiers.GO_TO_VERSION, sender: self)
```

Comment:
I would prefer not using segue ways because in long run one is not able to manipulate while presenting any viewcontroller

Reply:
This is true but sometimes a simple storyboard segue is good enough if you don't need to pass data or some other things. In this case, I think a storyboard segue to launch the Settings ViewController is not bad. But in other cases, I would recommend using code to instantiate the view controller.